recaf-ui/src/main/java/me/coley/recaf/ui/control/code/java/JavaArea.java

```diff
@@ -243,7 +243,10 @@ public SaveResult save() {
 			if (result.wasSuccess()) {
 				result.getValue().forEach((name, value) -> {
 					logger.info("Updating '{}'", name);
-					resources.getPrimary().getClasses().put(ClassInfo.read(value));
+					ClassSourceType type = ClassSourceType.PRIMARY;
+					if(getCurrentClassInfo() != null)
+						type = getCurrentClassInfo().getSourceType();
```

Comment:
Saving is disabled for non-primary classes, so this should never happen. Can inline to use constant `PRIMARY` value.